src/test/java/com/bobocode/svydovets/ioc/core/beanDefinitionFactory/BeanDefinitionFactoryTest.java

```diff
@@ -1,40 +1,83 @@
+import com.bobocode.svydovets.source.base.MessageService;
+import com.bobocode.svydovets.source.config.BasePackageWithAdditionalBeansConfig;
 import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import svydovets.core.context.beanDefinition.BeanAnnotationBeanDefinition;
+import svydovets.core.context.beanDefinition.BeanDefinition;
+import svydovets.core.context.beanDefinition.BeanDefinitionFactory;
+import svydovets.core.context.beanDefinition.ComponentAnnotationBeanDefinition;
+import svydovets.exception.BeanDefinitionCreateException;
+
+import java.util.Set;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
 
 class BeanDefinitionFactoryTest {
 
+    private BeanDefinitionFactory beanDefinitionFactory;
+
     @BeforeEach
     void setUp() {
+        beanDefinitionFactory = new BeanDefinitionFactory();
     }
 
     @AfterEach
     void tearDown() {
     }
 
     @Test
-    void registerBeanDefinitions() {
+    void shouldRegisterComponentAnnotationBeanDefinitions() {
+        beanDefinitionFactory.registerBeanDefinitions(Set.of(CommonService.class, MessageService.class));
+        BeanDefinition commonServiceBeandefinition = beanDefinitionFactory.getBeanDefinitionByBeanName("commonService");
+        BeanDefinition messageServiceBeanDefinition = beanDefinitionFactory.getBeanDefinitionByBeanName("messageService");
+        assertThat(commonServiceBeandefinition.getClass()).isEqualTo(ComponentAnnotationBeanDefinition.class);
+        assertThat(messageServiceBeanDefinition.getClass()).isEqualTo(ComponentAnnotationBeanDefinition.class);
+        assertThat(commonServiceBeandefinition).isNotNull();
+        assertThat(messageServiceBeanDefinition).isNotNull();
     }
 
     @Test
-    void registerBeanDefinition() {
+    void shouldRegisterComponentAnnotationBeanDefinition() {
+        beanDefinitionFactory.registerBeanDefinition(CommonService.class);
+        BeanDefinition commonServiceBeanDefinition = beanDefinitionFactory.getBeanDefinitionByBeanName("commonService");
+        assertThat(commonServiceBeanDefinition.getClass()).isEqualTo(ComponentAnnotationBeanDefinition.class);
+        assertThat(commonServiceBeanDefinition).isNotNull();
     }
 
     @Test
-    void getBeanDefinitionByBeanName() {
+    void shouldReturnBeanFromConfigClassWith() {
+        beanDefinitionFactory.registerBeanDefinitions(Set.of(CommonService.class, BasePackageWithAdditionalBeansConfig.class));
+        BeanDefinition trimServiceBeanDefinition = beanDefinitionFactory.getBeanDefinitionByBeanName("megaTrimService");
+        String scope = trimServiceBeanDefinition.getScope();
+        assertThat(trimServiceBeanDefinition.getClass()).isEqualTo(BeanAnnotationBeanDefinition.class);
+        assertThat(trimServiceBeanDefinition).isNotNull();
+        assertThat(scope).isEqualTo("singleton");
```

Comment:
Please, check if the "orderService" bean definition is created too. And I guess it may be better to create an additional test to check that all bean definition's fields is filled correct, not only "scope". Lines 59 and 62 can be removed